### PR TITLE
Migrate from F8 to Dekorate

### PR DESCRIPTION
**Pull request by Sgitario**
Instead of using fabric8 maven plugin to generate the resources, we'll now use dekorate and the dekorate test framework to run the test in a managed approach and unmanaged approach (deploying the app using s2i before running the test).

### CODE REVIEW

File: src/test/java/dev/snowdrop/example/UnmanagedOpenShiftIT.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+import org.junit.jupiter.api.condition.EnabledIfSystemProperty;
+
+import io.dekorate.testing.annotation.Inject;
+import io.dekorate.testing.openshift.annotation.OpenshiftIntegrationTest;
+import io.fabric8.kubernetes.api.model.Pod;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.LocalPortForward;
+
+@EnabledIfSystemProperty(named = "unmanaged-test", matches = "true")
+@OpenshiftIntegrationTest(deployEnabled = false, buildEnabled = false, pushEnabled = false)
```
Comment (Sgitario) [inline src/test/java/dev/snowdrop/example/UnmanagedOpenShiftIT.java:33]: I reported this issue https://github.com/dekorateio/dekorate/issues/801 to improve this.

File: pom.xml [hunk 2]
```diff
@@ -30,16 +30,10 @@
...
-    <maven-surefire-plugin.version>2.19.1</maven-surefire-plugin.version>
-    <maven-failsafe-plugin.version>2.19.1</maven-failsafe-plugin.version>
+    <maven-surefire-plugin.version>2.22.2</maven-surefire-plugin.version>
+    <maven-failsafe-plugin.version>2.22.2</maven-failsafe-plugin.version>
     <spring-boot.version>2.4.9</spring-boot.version>
-    <ubi.version>1.3</ubi.version>
-    <fabric8.generator.from>
-      registry.access.redhat.com/ubi8/openjdk-8:${ubi.version}
-    </fabric8.generator.from>
+    <dekorate.version>2.6.0</dekorate.version>
```
Comment (aureamunoz) [inline pom.xml:36]: `Dekorate` version should come from the parent, the  `snowdrop-dependencies`

**Review by aureamunoz — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 2 above)
Comment (Sgitario) [reply pom.xml]: The snowdrop-dependencies brings `dekorate:2.4.0` which for this repo is fine. Updated.

**Review by aureamunoz — APPROVED** [review]

**Comment by Sgitario** [comment]
I've updated the PR with deleting some Fabric8 leftovers and adding a missing annotation in the configuration.

**Review by iocanel — APPROVED** [review]
LGTM

**Review by aureamunoz — APPROVED** [review]
:+1:

File: src/main/resources/application.properties [hunk 3]
```diff
@@ -1 +1,14 @@
...
+dekorate.s2i.builder-image=registry.access.redhat.com/ubi8/openjdk-8:1.3
+## Labels
+dekorate.openshift.labels[0].key=app.kubernetes.io/part-of
+dekorate.openshift.labels[0].value=http-api-example
+dekorate.openshift.labels[1].key=app.kubernetes.io/component
+dekorate.openshift.labels[1].value=frontend
+dekorate.openshift.labels[2].key=app.openshift.io/runtime
+dekorate.openshift.labels[2].value=rh-spring-boot
+## Annotations
+dekorate.openshift.annotations[0].key=app.kubernetes.io/vcs-uri
```
[the remark this answers is not in the document]
Comment (Sgitario) [reply src/main/resources/application.properties]: Deleted

File: src/main/resources/application.properties [hunk 4]
```diff
@@ -1 +1,14 @@
...
+resteasy.jaxrs.defaultPath=/api
+
+# Dekorate
+dekorate.s2i.builder-image=registry.access.redhat.com/ubi8/openjdk-8:1.3
+## Labels
+dekorate.openshift.labels[0].key=app.kubernetes.io/part-of
+dekorate.openshift.labels[0].value=http-api-example
+dekorate.openshift.labels[1].key=app.kubernetes.io/component
+dekorate.openshift.labels[1].value=frontend
+dekorate.openshift.labels[2].key=app.openshift.io/runtime
```
[the remark this answers is not in the document]
Comment (Sgitario) [reply src/main/resources/application.properties]: Deleted

**Pull request merged**